ipython/dns/dns_suspicious_dendrogram.py

```diff
@@ -32,16 +33,19 @@ def main():
                 month=datetime.datetime.strptime(date[0], '%b').strftime('%m')
                 day=date[1]
                 ip_dst=row[2]
-                get_dendro(dbase,ip_dst,year,month,day,storage_path)
-
-def get_dendro(dbase,ip_dst,year,month,day,storage_path):
-
-    if not os.path.isfile("{0}dendro-{1}.csv".format(storage_path,ip_dst)):
-
-        dendro_qry = "hive -S -e \"set hive.cli.print.header=true; SELECT dns_a, dns_qry_name, ip_dst FROM (SELECT susp.ip_dst, susp.dns_qry_name, susp.dns_a FROM {0}.dns as susp  WHERE susp.y={1} AND susp.m={2} AND susp.d={3}  AND susp.ip_dst='{4}' ) AS tmp GROUP BY dns_a, dns_qry_name, ip_dst;\" | sed 's/[\\t]/,/g'> {5}dendro-{4}.csv".format(dbase,year,month,day,ip_dst,storage_path)
-
-        print dendro_qry
-        subprocess.call(dendro_qry,shell=True)
+                get_dendro(dbase,ip_dst,year,month,day,storage_path,impala_node)
+
+def get_dendro(dbase,ip_dst,year,month,day,storage_path,impala_node):
+
+    if not os.path.isfile("{0}dendro-{1}.csv".format(storage_path,ip_dst)):        
+        
+        dndro_qry = ("SELECT dns_a, dns_qry_name, ip_dst FROM (SELECT susp.ip_dst, susp.dns_qry_name, susp.dns_a FROM {0}.dns as susp WHERE susp.y={1} AND susp.m={2} AND susp.d={3}  AND susp.ip_dst=\"{4}\" ) AS tmp GROUP BY dns_a, dns_qry_name, ip_dst").format(dbase,year,month,day,ip_dst)           
+    
+        dendro_file = "{0}dendro-{1}.csv".format(storage_path,ip_dst)
+        impala_cmd = "impala-shell -i {0} --print_header -B --output_delimiter=',' -q '{1}' -o {2}".format(impala_node,dndro_qry,dendro_file)
+        
```

Comment:
could we add a hive_cmd and make it moduler for users who do not want to use impala?
something like:
`hive_cmd = "hive -S -e \"set hive.cli.print.header=true;" + dendro_qry + "| sed 's/[\\t]/,/g'>" + dendro_file`

with some sort of if statement checking a variable in the conf file for impala/hive